### PR TITLE
DDF-1205 complete guest interceptor support for dynamic policy handling

### PR DESCRIPTION
**Pull request by ryeats**
#### What does this PR do?
Complete guest interceptor support for dynamic policy handling
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@coyotesqrl @bcwaters @tbatie @roelens8 @jckilmer 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@pklinef
@stustison
#### How should this be tested?
Run the integration tests
#### Any background context you want to provide?
Guest interceptor did not support AsymmetricBinding policy 
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [x] Update / Add Integration Tests

### CODE REVIEW

File: catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/model/Search.java [hunk 1]
```diff
@@ -337,6 +337,9 @@ private void addObject(Map<String, Object> obj, String name, Object value) {
             actionJson.put(ACTIONS_TITLE, action.getTitle());
             actionJson.put(ACTIONS_DESCRIPTION, action.getDescription());
             actionJson.put(ACTIONS_URL, action.getUrl());
```
Comment (stustison) [inline catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/model/Search.java]: Do you need to remove this old setter for the action url.

Comment (ryeats) [inline catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/model/Search.java]: fixed

File: platform/security/interceptor/security-interceptor-guest/src/main/java/org/codice/ddf/security/interceptor/GuestInterceptor.java [hunk 2]
```diff
@@ -433,31 +242,24 @@ private void createSecurityToken(SoapVersion version, SOAPFactory soapFactory,
...
-
-                    } catch (SOAPException e) {
-                        LOGGER.error("Unable to convert SecurityToken to SOAPElement.", e);
-                    }
+                    securityToken = securityAssertion.getSecurityToken();
                 } else {
-                    LOGGER.warn("Subject did not contain a security assertion, could not assertion"
-                            + "to security header.");
+                    LOGGER.warn(
+                            "Subject did not contain a security assertion, could not assertion to security header.");
```
Comment (bcwaters) [inline platform/security/interceptor/security-interceptor-guest/src/main/java/org/codice/ddf/security/interceptor/GuestInterceptor.java]: It looks like this message is missing a word.
"Subject did not contain a security assertion, could not add assertion to SoapMessage."

**Comment by roelens8** [comment]
👍

**Comment by stustison** [comment]
👍

File: platform/security/interceptor/security-interceptor-guest/src/main/java/org/codice/ddf/security/interceptor/GuestInterceptor.java [hunk 3]
```diff
@@ -155,276 +111,129 @@ public GuestInterceptor(SecurityManager securityManager,
...
+        }
+
+        assertionInfoMap.entrySet()
+                .stream()
+                .flatMap(p -> p.getValue()
+                        .stream())
+                .filter(info -> MetadataConstants.ADDRESSING_ASSERTION_QNAME.equals(info.getAssertion()
+                        .getName()))
+                .findFirst()
+                .ifPresent(tru -> createAddressing(message, soapMessage));
```
Comment (coyotesqrl) [inline platform/security/interceptor/security-interceptor-guest/src/main/java/org/codice/ddf/security/interceptor/GuestInterceptor.java]: I'm not a fan of the side-effect use of `ifPresent`. If you're not going to need the filtered object from the optional, I'd as soon see a terminating call to `isPresent` instead to get a boolean. Something like...
```
boolean hasAddressingAssertion = ...
    .isPresent();
if (hasAddressingAssertion) {
    createAddressing(message, soapMessage);
}
```

**Comment by pklinef** [comment]
It does not look like you added any new integration tests that would test symmetric and asymmetric bindings.  I think our current itests only test transport binding.

**Comment by stustison** [comment]
@pklinef has a good point. We do have a sample SOAP endpoint in the SDK and we could add policies for the other two bindings into its WSDL and test out those other policies. That endpoint is already being deployed during the itests, so it should be relatively easy. You'd just need to try the same things were already doing against the endpoints associated with the other two policies. (it currently just has a TransportBinding policy in it)

**Comment by pklinef** [comment]
👍  other than itests.

**Comment by coyotesqrl** [comment]
Hero build 👍

**Pull request merged**